### DISCUSSION TITLE
PID algorithm does not read sensor every second, but current value is limited to teleperiod.

### DISCUSSION THREAD

**Discussion by Rush56** [arendst/Tasmota#22275] [Support request]
In GUI i can see sensor updating frequently, but current value - pidPV only gets updated on teleperiod intervals. I made an automation based on required Power output from PID via voltage regulator for some heating/sterilizing application, which is working fine, but already RTD sensor is quite slow, but still managable.

I have defined PID_USE_LOCAL_SENSOR and have set PID_UPDATE_SECS to 0, which should take readings when new data is avaliable. But since it is still limited to teleperiod 10, which is lowest i can set it does mess up a bit with automation and is not consistant.

I've found some similar issues:
https://github.com/arendst/Tasmota/issues/17636
So i tryed adding this line in driver, but does not change anything:
https://github.com/arendst/Tasmota/commit/dad059737d34015eeae594211d17bf6250866c79

**Comment by Wintespe** [comment]
That's the new problem I suspected here:

PID PID_LOCAL_SENSOR_NAME doesn't seem to update Pv #22094

I haven't done any further analysis yet because my somewhat older version gets an update of the temperature data every second without any problems. I use Tasmota Version 11.1.0  This version does not have the change from #17636 . Try one of the newer versions that is close to it.

**Reply by Rush56** [reply]
Ok thanks. It doesn't cause me too much trouble with process, but if it will be needed to fix then i will have to try older versions.

I tryed disabling local_sensor and creating a rule which picks sensor data and sets pidpv, based on console output that was performing every second or so. While that seemed to update sensor data 2-3x faster in practice based on GUI data, but in that case the more the temperature rose the more delay it had from actual data. 
It was much more reliable on teleperiod 10 then, since then on every 10 seconds it got correct value with minimal delay.
